articles/object-arrays-destructuring/index.md

```diff
@@ -0,0 +1,252 @@
+
+```javascript
+var person = {};
+
+person.1 = "Head";
+```
+
+The correct way of adding a property using an invalid identifier is by bracket notation.
+
+```javascript
+var person = {};
+
+person[1] = "Head";
+```
+
+When you use brackets, the invalid identifiers are stringified. Try adding a property with a `string` key to the `person` object using brackets.
+
+```javascript
+var person = {};
+
+person[hobby] = "coding";
+```
+When you execute the code above you get an error. `Uncaught ReferenceError: hobby is not defined
+`. This is because anything without quotation marks in JavaScript should be either: 
+- a JavaScript keyword
+- number
+- boolean
+- defined somewhere. 
+
+You can therefore solve the error by either defining `hobby` or using by simply using quotation marks.
+
+```javascript
+// by defining hobby
+var person = {};
+var hobby = "myHobby";
+
+person[hobby] = "coding"; 
+```
+
+```javascript
+// by using quotation marks
+var person = {};
+
+person["hobby"] = "coding";
+```
+
+All the keys in the square bracket notation except symbols are converted to strings. This is because objects property names in JavaScript should be a string or a symbol. 
+
+### Accessing object properties
+        
+Objects properties can be accessed by the use of **dot notation** or **square brackets notation**. 
+
+```javascript
+// access using dot nation
+person.name;
+// access using bracket notation
+person["name"];
+```
+
+> Properties with multi-word keys can't be accessed using the dot notation. They can only be accessed using bracket notation.
+
+Properties in objects can be deleted using the `delete` operator.
+
+```javascript
+delete person.role;
+```
+
+### 2. Arrays
+
+Arrays are variables used to store multiple items and ordered collections. They use zero-based indexing meaning that the first item in the array is index 0. Arrays can be created using the `var`, `let` or `const` keywords.
+
+```javascript
+var fruits = [
+	"Bananas",
+	"Lemons",
+	"Mangoes"
+];
+```
+
+In arrays, unlike objects, keys are auto-generated. Each item is given a numeric key according to the position in the array.
+
+### Adding items to arrays
+
+Items can be added at the beginning, between and the end of the array. Items can be added at the beginning of an array by using [unshift()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift) method.
+
+```javascript
+fruits.unshift("Watermelon","Plums");
+```
+
+The unshift() method adds the items at the beginning of the `fruits` array and returns the length of the array. The first item in your unshift method will be first in the resultant array. 
+
+To add items in between other items, [splice()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) method is used. The index of the items before and after the position you want to add the item is passed in the splice() method.
```

Comment:
While your code will place "oranges" between the first and second fruit in the array, the indexes are not the index of the items before and after the position that you would want to add the item. Instead, the first argument is the position at which insertion/deletion is to begin and the second is the number of elements that should be deleted.